Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -647,4 +657,52 @@ public class PBXProjGenerator {
         groups.insert(group, at: 0)
         return (allSourceFiles, groups)
     }
+
+    func getSourceChildren(sourceMetadata source: Source, dirPath: Path) throws -> [Path] {
+        let excludedFiles = [".DS_Store"].map { dirPath + Path($0) }
+
+        let sourcePath = Path(source.path)
+
+        /*
+            Exclude following if mentioned in Source.excludes.
+            Any path related to Source base path
+            + Any path related to current dirPath
+            + Pre-defined Excluded files
+        */
+
+        let sourceExcludeFilePaths: Set<Path> = Set(getSourceExcludes(sourceMetadata: source, dirPath: sourcePath)
+                + getSourceExcludes(sourceMetadata: source, dirPath: dirPath)
+                + excludedFiles)
+
+        return try dirPath.children().sorted()
+        .filter {
```

Comment:
If we sort after we filter it's technically faster. Probably pre-mature optimization, but maybe someone has thousands of files